OpenTrons/opentrons-api
# fix(protocol-designer): fix pipette type button display condition in EditInstrumentsModal

# Overview
This PR fixes multiple bugs in `EditInstrumentsModal`

1. pipette type button display issue.
2. Add a pipette button issue
3. swap button display condition
4. buttons for the modal

### after

https://github.com/user-attachments/assets/b7b6afbd-9ce3-44e0-b6b3-25cc5a7bf01b



close RQA-3285 RQA-3294 RQA-3296


## Test Plan and Hands on Testing
Need to check the following cases

RQA-3285 
1. create or import a protocol that has only one pipette

2. click instruments edit button

4. click edit button in the existing pipette
check there is 96-channel button then hit cancel

5. click add a pipette button
check there is no 96-channel button (because of the existing pipette)

6. add a pipette single or multi

7. click edit for each mount
check there is no 96-channel button 

8. remove both mounts and click a pipette button 
there is 96-channel button (2 add a pipette buttons will show up which is a bug and it will be resolved by a following pr) 

Add a pipette button
only one button is always placed below a pipette info card
when clicking swap, add a pipette button is placed the same position




## Changelog
- add a new util function and its tests
- update text in Edit Instruments modal and test



## Review requests



## Risk assessment
low
